server/src/actors/outcome.rs

```diff
@@ -80,45 +80,71 @@ pub enum Outcome {
     Duplicate,
+
+    /// [Relay] There was no valid project id in the request or the required project does not exist.
     ProjectId,
+
+    /// [Relay] The protocol version sent by the SDK is not supported and parts of the payload may
+    /// be invalid.
     AuthVersion,
+
+    /// [Legacy] The SDK did not send a client identifier.
+    ///
+    /// In Relay, this is no longer required.
     AuthClient,
+
+    /// [Relay] The store request was missing an event payload.
     NoData,
+
+    /// [Relay] The event payload exceeds the maximum size limit for the respective endpoint.
     TooLarge,
+
+    /// [Legacy] A store request was received with an invalid method.
+    ///
+    /// This outcome is no longer emitted by Relay, as HTTP method validation occurs before an event
+    /// id or project id are extracted for a request.
     DisallowedMethod,
+
+    /// [Relay] The content type for a specific endpoint did not match the whitelist.
+    ///
+    /// While the standard store endpoint allows all content types, other endpoints may have
+    /// stricter requirements.
     ContentType,
+
+    /// [Legacy] The project id in the URL does not match the one specified for the public key.
+    ///
+    /// This outcome is no longer emitted by Relay. Instead, Relay will emit a standard `ProjectId`
+    /// since it resolves the project first, and then checks for the valid project key.
     MultiProjectId,
+
+    /// [Relay] A minidump file was missing for the minidump endpoint.
     MissingMinidumpUpload,
+
+    /// [Relay] The file submitted as minidump is not a valid minidump file.
     InvalidMinidump,
+
+    /// [Relay] The security report was not recognized due to missing data.
     SecurityReportType,
-    SecurityReport,
-    Cors,
 
-    // Outcomes only emitted by Relay, not in Sentry
-    // ---------------------------------------------
-    ///  The event payload exceeds the maximum size limit for the respective endpoint.
-    PayloadTooLarge,
+    /// [Relay] The security report did not pass schema validation.
+    SecurityReport,
 
-    /// The payload length was not specified in headers.
-    UnknownPayloadLength,
+    /// [Relay] The request origin is not allowed for the project.
+    Cors,
 
-    /// Reading or decoding the payload from the socket failed for any reason.
-    InvalidPayloadFormat,
+    /// [Relay] Reading or decoding the payload from the socket failed for any reason.
+    Payload,
 
-    /// Parsing the event JSON payload failed due to a syntax error.
-    InvalidPayloadJsonError,
+    /// [Relay] Parsing the event JSON payload failed due to a syntax error.
+    InvalidJson,
 
-    /// A project state returned by the upstream could not be parsed.
+    /// [Relay] A project state returned by the upstream could not be parsed.
     ProjectState,
 
-    /// Relay does not support the protocol version sent by the SDK and may not understand the event
-    /// payload.
-    UnsupportedProtocolVersion,
```

Comment:
This could be useful as a stand-alone code, it is probably not an internal error ( it is probably a client error) and we could use it to detect old/buggy clients.